README.md

```diff
@@ -32,28 +36,79 @@ pedlar.perform(() => {
   return () => console.log('Side effect 1 cleaned up!')
 })
 
+// LOG: 'Side effect 1 performed'
+
 pedlar.perform(() => {
   console.log('Side effect 2 performed')
   return () => console.log('Side effect 2 cleaned up!')
 })
 
+// LOG: 'Side effect 2 performed'
+
+pedlar.destroyAll()
+
+// LOG: 'Side effect 1 cleaned up!'
+// LOG: 'Side effect 2 cleaned up!'
+```
+
+### Add and automatically remove an event
+
+```js
+let pedlar = new Pedlar()
+let el = document.getElementById('my-button')
+
+// Adds the click event
+pedlar.addEvent(el, 'click', () => console.log('My button was clicked'))
+
+// ...
+
+// Removes the click event, along with cleaning up all other side effects
 pedlar.destroyAll()
 ```
 
+### Re-perform an event when dependencies have changed
+
+```js
+let pedlar = new Pedlar()
+let result = pedlar.perform(() => {
+  console.log('Side effect performed')
+  return () => console.log('Side effect cleaned up!')
+}, ['my-dependency'])
+
+// LOG: 'Side effect performed'
+
+result.performAgain(['my-dependency'])
```

Comment:
calling this does not actually perform the effect again unless the dependency you pass in has changed?

Reply:
That's correct.  I had originally called this "performAgain" to distinguish it from the root "perform" function, but I have since changed it to also just use "perform".  

Do you think that calling it just "perform" resolves this confusion?  You are allowed to just call `result.perform()` without and dependencies and it will run the side effect.